repair/repair.cc

```diff
@@ -934,211 +933,6 @@ repair_neighbors repair_info::get_repair_neighbors(const dht::token_range& range
         neighbors[range];
 }
 
-// Repair a single cf in a single local range.
-// Comparable to RepairJob in Origin.
-static future<> repair_cf_range(repair_info& ri,
-        sstring cf, ::dht::token_range range,
-        const std::vector<gms::inet_address>& neighbors) {
-    if (neighbors.empty()) {
-        // Nothing to do in this case...
-        return make_ready_future<>();
-    }
-
-    ri.check_in_abort();
-    return estimate_partitions(ri.db, ri.keyspace, cf, range).then([&ri, cf, range, &neighbors] (uint64_t estimated_partitions) {
-    range_splitter ranges(range, estimated_partitions, ri.target_partitions);
-    return do_with(seastar::gate(), true, std::move(cf), std::move(ranges),
-        [&ri, &neighbors] (auto& completion, auto& success, const auto& cf, auto& ranges) {
-        return do_until([&ranges] () { return !ranges.has_next(); },
-            [&ranges, &ri, &completion, &success, &neighbors, &cf] () {
-            auto range = ranges.next();
-            check_in_shutdown();
-            ri.check_in_abort();
-            return seastar::get_units(parallelism_semaphore, 1).then([&ri, &completion, &success, &neighbors, &cf, range] (auto signal_sem) {
-                auto checksum_type = repair_checksum::streamed;
-
-                // Ask this node, and all neighbors, to calculate checksums in
-                // this range. When all are done, compare the results, and if
-                // there are any differences, sync the content of this range.
-                std::vector<future<partition_checksum>> checksums;
-                checksums.reserve(1 + neighbors.size());
-                checksums.push_back(checksum_range(ri.db, ri.keyspace, cf, range, checksum_type));
```

Comment:
Surely, more code can be dropped - checksum_range for example, and the verbs in messaging_service.

@asias please follow up with more code destruction.

Reply:
I will do the remaining after this patch set.